src/base/snark0.ml

```diff
@@ -77,387 +80,7 @@ struct
-              let open Cvar in
-              assert_r1cs (a + a) b (a + b - c)
-            in
-            res
-
-      module Array = struct
-        let num_true (bs : var array) =
-          Array.fold bs ~init:(Cvar.constant Field.zero) ~f:(fun x y ->
-              Cvar.add x (y :> Cvar.t) )
-
-        let any = function
-          | [||] ->
-              return false_
-          | [| b1 |] ->
-              return b1
-          | [| b1; b2 |] ->
-              b1 || b2
-          | bs ->
-              let open Let_syntax in
-              let%map all_zero =
-                equal (num_true bs) (Cvar.constant Field.zero)
-              in
-              not all_zero
-
-        let all = function
-          | [||] ->
-              return true_
-          | [| b1 |] ->
-              return b1
-          | [| b1; b2 |] ->
-              b1 && b2
-          | bs ->
-              equal
-                (Cvar.constant (Field.of_int (Array.length bs)))
-                (num_true bs)
-
-        module Assert = struct
-          let any bs = assert_non_zero (num_true bs)
-
-          let all bs =
-            assert_equal (num_true bs)
-              (Cvar.constant (Field.of_int (Array.length bs)))
-        end
-      end
-
-      let equal (a : var) (b : var) = a lxor b >>| not
-
-      let of_field x =
-        let open Let_syntax in
-        let%map () = assert_ (Constraint.boolean x) in
-        create x
-
-      module Unsafe = struct
-        let of_cvar (t : Cvar.t) : var = create t
-      end
-
-      module Assert = struct
-        let ( = ) (x : var) (y : var) = assert_equal (x :> Cvar.t) (y :> Cvar.t)
-
-        let is_true (v : var) = v = true_
-
-        let%snarkydef_ any (bs : var list) =
-          assert_non_zero (Cvar.sum (bs :> Cvar.t list))
-
-        let%snarkydef_ all (bs : var list) =
-          assert_equal
-            (Cvar.sum (bs :> Cvar.t list))
-            (Cvar.constant (Field.of_int (List.length bs)))
-
-        let%snarkydef_ exactly_one (bs : var list) =
-          assert_equal (Cvar.sum (bs :> Cvar.t list)) (Cvar.constant Field.one)
-      end
-
-      module Expr = struct
-        type t = Var of var | And of t list | Or of t list | Not of t
-
-        let rec eval t =
-          let open Let_syntax in
-          match t with
-          | Not t ->
-              eval t >>| not
-          | Var v ->
-              return v
-          | And ts ->
-              Checked.all (List.map ~f:eval ts) >>= all
-          | Or ts ->
-              Checked.all (List.map ~f:eval ts) >>= any
-
-        let assert_ t = eval t >>= Assert.is_true
-
-        let ( ! ) v = Var v
-
-        let ( && ) x y = And [ x; y ]
-
-        let ( &&& ) = ( && )
-
-        let ( || ) x y = Or [ x; y ]
-
-        let ( ||| ) = ( || )
-
-        let not t = Not t
-
-        let any xs = Or xs
-
-        let all xs = And xs
-      end
-    end
+    include Utils.Make (Backend) (Checked) (As_prover_old) (Runner)
```

Comment:
Looks like you could use the `As prover` module directly here or am I mistaken?

Reply:
Indeed, I was trying to solve another thing and it's no longer necessary.